src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java

```diff
@@ -2495,8 +2543,17 @@ public void registerOutParameter(String parameterName, int sqlType, String typeN
             loggerExternal.entering(getClassNameLogging(), "registerOutParameter",
                     new Object[] {parameterName, sqlType, typeName});
         checkClosed();
-        registerOutParameter(findColumn(parameterName, CallableStatementGetterSetterMethod.IS_SETTER_METHOD), sqlType,
-                typeName);
+
+        int index = findColumn(parameterName, CallableStatementGetterSetterMethod.IS_SETTER_METHOD);
+
+        if (loggerExternal.isLoggable(java.util.logging.Level.FINER))
+            loggerExternal.entering(getClassNameLogging(), "registerOutParameter",
+                    new Object[] {index, sqlType, typeName});
+
+        checkClosed();
```

Comment:
How likely is it that the connection is open before findColumn() and closed after? +2 more instances of the same below.